lib/active_record/connection_adapters/postgis/postgis_database_tasks.rb

```diff
@@ -96,11 +96,11 @@ def extension_names
         end
 
         def ensure_installation_configs
-          if configuration["setup"] == "default" && !configuration["postgis_extension"]
+          if configuration_hash[:setup] == "default" && !configuration_hash[:postgis_extension]
             share_dir = `pg_config --sharedir`.strip rescue "/usr/share"
             control_file = ::File.expand_path("extension/postgis.control", share_dir)
             if ::File.readable?(control_file)
-              configuration["postgis_extension"] = "postgis"
+              @configuration_hash = configuration_hash.merge(postgis_extension: "postgis")
```

Comment:
frozen

Reply:
We should freeze the resultant hash since this assignment undoes it.

```rb
h1 = {a: 1, b: 2}.freeze
p h1.frozen?
#=> true

h2 = h1.merge({c:3})
p h2.frozen?
#=> false
```

It is probably not a big deal, but I'd rather not have this be the source of weird downstream errors

Reply:
Good point. Added `.freeze`